packages/server/src/question/question-role.guard.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  Injectable,
+  UnauthorizedException,
+  NotFoundException,
+} from '@nestjs/common';
+import { UserModel } from '../profile/user.entity';
+import { QuestionModel } from './question.entity';
+import { QueueModel } from '../queue/queue.entity';
+import { RolesGuard } from '../guards/role.guard';
+
+@Injectable()
+export class QuestionRolesGuard extends RolesGuard {
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+  async setupData(
+    request: any,
+  ): Promise<{ courseId: number; user: UserModel }> {
+    let queueId = request.params.queueId;
```

Comment:
If QuestionRolesGuard is only for question controller, in what cause is there actually a `queueId` in the params?

Reply:
This is specifically for the create question endpoint. (if I remember correctly) Because in that case we don't have a preexisting question id, but we do have the queueid

Reply:
Ah i see. might be better to just do the authorization directly in the endpoint for something like that then

Reply:
I wanted I wanted to keep it because there was alot of repeated code within each of the endpoints which is stopped by the guard, but definitely something to consider.